File: s03_Java/CountDownTimer/app/src/main/java/com/omeryildizce/countdowntimer/MainActivity.java
package com.omeryildizce.countdowntimer;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Color;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView textView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);
        new CountDownTimer(10_000, 1_000){
            @Override
            public void onTick(long millisUntilFinished) {
                textView.setText(String.format("Left: %d", millisUntilFinished / 1000));
            }

            @Override
            public void onFinish() {
                textView.setText("Left: "+0);
                textView.setTextColor(Color.RED);
            }
        }.start();
    }
}

File: s16_Kotlin/CountDown/app/src/main/java/com/omeryildizce/countdown/MainActivity.kt
package com.omeryildizce.countdown

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import com.omeryildizce.countdown.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        object : CountDownTimer(59_000, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                println(millisUntilFinished)
                binding.counter.setText(String.format("%02d:%02d:%02d", 0, 0, (millisUntilFinished/1000).toInt()))
            }

            override fun onFinish() {
                binding.counter.setText("Time over")

            }

        }.start()

    }
}